### ISSUE TITLE
schema[playbook] failures when using a variable with `order` keyword

### ISSUE THREAD

**Issue by jhongturney** [ansible/ansible-lint#4061]
##### Summary

In a handful of playbooks we have an order keyword defined like so, so that we have the option of passing in a value for the variable at runtime, which has been working just fine:

> order: "{{ host_order | default('shuffle') }}"

`ansible-lint`, however, returns this unskippable failure:

> WARNING  Listing 1 violation(s) that are fatal
schema[playbook]: $[1].order "{{ host_order | default('shuffle') }}" is not one of ['default', 'sorted', 'reverse_sorted', 'reverse_inventory', 'shuffle']

Ultimately the field is populated with valid value, but the checking here is too strict, or at least should be skippable.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```
ansible-lint 24.2.0 using ansible-core:2.16.4 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
